src/utils/resolve_url.ts

```diff
@@ -122,4 +68,192 @@ function getFilenameIndexInUrl(url: string): number {
   return indexOfLastSlash + 1;
 }
 
+/**
+ * Resolve the output URL from the baseURL and the relative reference as
+ * specified by RFC 3986 section 5.
+ * @param base
+ * @param relative
+ * @see https://datatracker.ietf.org/doc/html/rfc3986#section-5
+ * @example base: http://example.com | relative: /b/c | output: http://example.com/b/c
+ * @returns the resolved url
+ */
+function _resolveURL(base: string, relative: string) {
+  const baseParts = parseURL(base);
+  const relativeParts = parseURL(relative);
```

Comment:
Why is it plural here?

Reply:
I'm dumb, reviewed in a too-fast fashion, removing this now that I actually took some time and realized that my comment was nonsensical